### DISCUSSION TITLE
Is it possible to include whole venv/conda environment?

### DISCUSSION THREAD

**Discussion by ladi-pomsar** [pyinstaller/pyinstaller#8589] [Help]
Hello,

I am doing some dynamic importing of scripts and end up lacking dependencies. I have all the needed dependencies in my venv, but after the pyInstaller does the packaging, it understandably doesn't recognize the dependencies of dynamic imports and ends up lacking them in a runtime. 

Is it possible to somehow easily include all libraries installed in venv to .exe? Would this also work for conda environment? I am working with windows, so there may also be some .dlls needed to be included.

Thanks

**Comment by rokm** [comment]
> Is it possible to somehow easily include all libraries installed in venv to .exe?

Nope, because this goes against PyInstaller's packaging philosophy. But if you really want to do this, you can iterate over all importable modules (`pkgutil.iter_modules`) and add them to hidden imports at the start of your .spec file, and then pass that list to Analysis.

If you have a relatively small set of those imports, though, it might be easier to specify them as hidden imports directly.

> I am working with windows, so there may also be some .dlls needed to be included.

If those DLLs are dynamically imported from code, we might or might not be able to pick them up automatically (depending on how exactly they are imported). If they are link-time dependencies of for example collected python binary extensions, then they should be collected automatically.
